### PR TITLE
prototype for easier mapping

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Working on a new approach for simplifying the parameterization cases

### CODE REVIEW

File: results/results_timestamp-2024-04-15-16-27-38_machine-Codys-MBP.attlocal.net_environment-819e811ae0d4d7b81932109b976c94c7b8e7545d.json
Comment (CodyCBakerPhD) [inline results/results_timestamp-2024-04-15-16-27-38_machine-Codys-MBP.attlocal.net_environment-819e811ae0d4d7b81932109b976c94c7b8e7545d.json:1]: Adding note here that I really should indicate somehow this was on the DataJoint WiFi

**Comment by CodyCBakerPhD** [comment]
@oruebel This is good to go - the sooner we get it through the fewer conflicts we'll have to fix here

For newly added tests or tests on other PRs, they can either use the previous structure or upgrade to the new one

**Review by oruebel — APPROVED** [review]
Looks good to me. I'll need to take a stab at updating the PR's for Zarr and LINDI/

**Comment by oruebel** [comment]
@sinha-r this PR changes how we define the parametrize of tests to follow the design we discussed previously. I.e., we now have a dictionary, where each entry is a particular case with a set of parameters instead of having separate lists for each parameter we are specifying. 


```
parameter_cases = dict(
    IBLTestCase1=dict(
        s3_url=get_s3_url(dandiset_id="000717", dandi_path="sub-mock/sub-mock_ses-ecephys1.nwb"),
    ),
    ClassicRos3TestCase=dict(s3_url="https://dandiarchive.s3.amazonaws.com/ros3test.nwb"),
)
```

**Comment by oruebel** [comment]
> For newly added tests or tests on other PRs, they can either use the previous structure or upgrade to the new one

I updated #51 and #53  to use the new structure

**Pull request merged**